sdk/eventhub/Azure.Messaging.EventHubs/design/api-skeleton-january-2020.md

```diff
@@ -0,0 +1,535 @@
+{
+    public TimeSpan? MaximumWaitTime { get; set; }
+    public long? OwnerLevel { get; set; }
+    public bool TrackLastEnqueuedEventProperties { get; set; }
+}
+```
+
+### `Azure.Messaging.EventHubs.Producer`
+```csharp
+public class EventHubProducerClient : IAsyncDisposable 
+{
+    public bool IsClosed { get; protected set; }
+    public virtual string EventHubName { get; }
+    public virtual string FullyQualifiedNamespace { get; }
+    
+    // Basic constructors
+    public EventHubProducerClient(string connectionString);
+        
+    public EventHubProducerClient(
+        string connectionString, 
+        EventHubProducerClientOptions clientOptions);
+        
+    public EventHubProducerClient(
+        string connectionString, 
+        string eventHubName);
+        
+    public EventHubProducerClient(
+        string connectionString, 
+        string eventHubName, 
+        EventHubProducerClientOptions clientOptions);
+    
+    // Advanced constructors
+    public EventHubProducerClient(
+        string fullyQualifiedNamespace, 
+        string eventHubName, 
+        TokenCredential credential, 
+        EventHubProducerClientOptions clientOptions = null);
+        
+    public EventHubProducerClient(
+        EventHubConnection connection, 
+        EventHubProducerClientOptions clientOptions = null);
+    
+    // Mocking constructors
+    protected EventHubProducerClient();
+    
+    public virtual Task CloseAsync(CancellationToken cancellationToken = default);
+    public virtual ValueTask DisposeAsync();
+    
+    public virtual Task<EventHubProperties> GetEventHubPropertiesAsync(CancellationToken cancellationToken = default);
+    public virtual Task<string[]> GetPartitionIdsAsync(CancellationToken cancellationToken = default);
+    public virtual Task<PartitionProperties> GetPartitionPropertiesAsync(string partitionId, CancellationToken cancellationToken = default);
+    
+    public virtual ValueTask<EventDataBatch> CreateBatchAsync(CreateBatchOptions options, CancellationToken cancellationToken = default);
+    public virtual ValueTask<EventDataBatch> CreateBatchAsync(CancellationToken cancellationToken = default);
+    
+    public virtual Task SendAsync(EventDataBatch eventBatch, CancellationToken cancellationToken = default);
+}
+
+public class EventHubProducerClientOptions 
+{
+    public EventHubConnectionOptions ConnectionOptions { get; set; }
+    public EventHubsRetryOptions RetryOptions { get; set; }
+}
+
+public class CreateBatchOptions
+{
+    public long? MaximumSizeInBytes { get; set; }
+    public string PartitionId { get; set; }
+    public string PartitionKey { get; set; }
+}
+
+public sealed class EventDataBatch : IDisposable 
+{
+    public int Count { get; }
+    public long MaximumSizeInBytes { get; }
+    public long SizeInBytes { get; }
+    public void Dispose();
+    public bool TryAdd(EventData eventData);
+}
+```
+
+### `Azure.Messaging.EventHubs.Processor`
```

Comment:
`ProcessEventArgs`, `PartitionInitializingEventArgs`, `PartitionClosingEventArgs`..etc classes/structs come under `Azure.Messaging.EventHubs` package. 

I think we should remove `Azure.Messaging.EventHubs.Processor` header from here.

Reply:
That header is the namespace, which exists in the core Event Hubs package and is intended to contain types common between different opinionated processor implementations (such as blob storage, Redis, Cosmos, etc).